src/uct/tcp/tcp_ep.c

```diff
@@ -230,26 +220,11 @@ static UCS_CLASS_INIT_FUNC(uct_tcp_ep_t, uct_tcp_iface_t *iface,
     ucs_queue_head_init(&self->pending_q);
     ucs_queue_head_init(&self->put_comp_q);
 
-    /* Make a socket non-blocking if an EP is created during accepting
-     * a connection or non-blocking connection mode is requested */
-    if ((self->fd != -1) && ((dest_addr == NULL) || iface->config.conn_nb)) {
-        status = ucs_sys_fcntl_modfl(self->fd, O_NONBLOCK, 0);
-        if (status != UCS_OK) {
-            goto err_cleanup;
-        }
-    }
-
-    /* Set socket options if the socket fd was created, i.e. EP was created
-     * using CONNECT_TO_IFACE method or it is EP created during accepting
-     * remote connection */
-    if (self->fd != -1) {
-        status = uct_tcp_iface_set_sockopt(iface, self->fd);
-        if (status != UCS_OK) {
-            goto err_cleanup;
-        }
-
+    if (self->fd != -1) /* EP is created during accepting a connection */ {
         self->conn_retries++;
     } else if (dest_addr == NULL) {
+        /* Since no socket FD and no destination address were specified for
+         * new EP, it means that EP is created with CONNECT_TO_EP method */
```

Comment:
is there some EP flag we can check instead?

Reply:
this is a constructor for `EP`, we don't have any flags initialized here